pkg/services/authz/zanzana/client/client.go

```diff
@@ -0,0 +1,188 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/authz/zanzana/schema"
+)
+
+type ClientOption func(c *Client)
+
+func WithTenantID(tenantID string) ClientOption {
+	return func(c *Client) {
+		c.tenantID = tenantID
+	}
+}
+
+func WithLogger(logger log.Logger) ClientOption {
+	return func(c *Client) {
+		c.logger = logger
+	}
+}
+
+type Client struct {
+	logger   log.Logger
+	client   openfgav1.OpenFGAServiceClient
+	tenantID string
+	storeID  string
+	modelID  string
+}
+
+func New(ctx context.Context, cc grpc.ClientConnInterface, opts ...ClientOption) (*Client, error) {
+	c := &Client{
+		client: openfgav1.NewOpenFGAServiceClient(cc),
+	}
+
+	for _, o := range opts {
+		o(c)
+	}
+
+	if c.logger == nil {
+		c.logger = log.NewNopLogger()
+	}
+
+	if c.tenantID == "" {
+		// TODO what should be the default value?
+		c.tenantID = "default"
```

Comment:
Are we namespacing as in stackid:23 for when it's used?

If so I guess we could do stackid:default

Reply:
Yeah this is a tricky one. I think for stacks it probably should match namespaces, e.g. `stack-<id>`.